package/thingino-esphome/plugins/thingino_media_player/thingino_media_player.c

```diff
@@ -0,0 +1,1481 @@
+        audio_input_callback_t callback = g_player_ctx.audio_input_callback;
+        void *userdata = g_player_ctx.audio_input_userdata;
+        pthread_mutex_unlock(&g_player_ctx.lock);
+
+        if (!active || sockfd < 0) {
+            break;
+        }
+
+        bytes_received = read(sockfd, buffer, AUDIO_INPUT_BUFFER_SIZE);
+        if (bytes_received <= 0) {
+            if (bytes_received < 0) {
+                perror("[MediaPlayer] read from audio input socket");
+            }
+            break;
+        }
+
+        // Call the callback with the audio data
+        if (callback) {
+            // Convert bytes to samples (16-bit = 2 bytes per sample)
+            size_t samples = bytes_received / 2;
+            callback((const int16_t *)buffer, samples, userdata);
+        }
+    }
+
+    printf("[MediaPlayer] Audio input thread exiting\n");
+
+    pthread_mutex_lock(&g_player_ctx.lock);
+    if (g_player_ctx.audio_input_sock >= 0) {
+        close(g_player_ctx.audio_input_sock);
+        g_player_ctx.audio_input_sock = -1;
+    }
+    g_player_ctx.audio_input_active = false;
+    pthread_mutex_unlock(&g_player_ctx.lock);
+
+    return NULL;
+}
+
+int media_player_start_audio_input(audio_input_callback_t callback, void *userdata) {
+    if (!callback) {
+        fprintf(stderr, "[MediaPlayer] Audio input callback is NULL\n");
+        return -1;
+    }
+
+    pthread_mutex_lock(&g_player_ctx.lock);
+
+    // Stop any existing audio input
+    if (g_player_ctx.audio_input_active && g_player_ctx.audio_input_sock >= 0) {
+        close(g_player_ctx.audio_input_sock);
+        g_player_ctx.audio_input_sock = -1;
+        g_player_ctx.audio_input_active = false;
+    }
+
+    // Connect to audio input socket
+    int sockfd = iac_connect_audio_input();
+    if (sockfd < 0) {
+        pthread_mutex_unlock(&g_player_ctx.lock);
+        return -1;
+    }
+
+    // Send audio input request
+    int request_type = AUDIO_INPUT_REQUEST;
+    if (write(sockfd, &request_type, sizeof(int)) != sizeof(int)) {
+        perror("[MediaPlayer] write audio input request");
+        close(sockfd);
+        pthread_mutex_unlock(&g_player_ctx.lock);
+        return -1;
+    }
+
+    g_player_ctx.audio_input_sock = sockfd;
+    g_player_ctx.audio_input_callback = callback;
+    g_player_ctx.audio_input_userdata = userdata;
+    g_player_ctx.audio_input_active = true;
+
+    pthread_mutex_unlock(&g_player_ctx.lock);
+
+    // Start the audio input thread
+    if (pthread_create(&g_player_ctx.audio_input_thread, NULL, audio_input_thread, NULL) != 0) {
+        perror("[MediaPlayer] pthread_create for audio input");
+        pthread_mutex_lock(&g_player_ctx.lock);
+        close(g_player_ctx.audio_input_sock);
+        g_player_ctx.audio_input_sock = -1;
+        g_player_ctx.audio_input_active = false;
+        pthread_mutex_unlock(&g_player_ctx.lock);
+        return -1;
+    }
+
+    printf("[MediaPlayer] Audio input started (16kHz 16-bit mono)\n");
```

Comment:
We broadcast support for other audio sample rates but only assume support for the preconfigured sample rate in IAD. We should either trim our broadcast supported sample rates or have some logic to resample the audio to what IAD needs.